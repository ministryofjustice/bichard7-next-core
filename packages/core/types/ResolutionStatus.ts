enum ResolutionStatus {
  READONLY = 4, // Intermediate state when a trigger is unresolved but the current user doesn't handle the trigger. Doesn't get written to the database (from legacy bichard)
  RESOLVED = 2,
  SELECTED = 5, // Intermediate state when the checkbox for a trigger has been selected but the 'Mark Selected As Complete' has not been pressed. Doesn't get written to the database
  SUBMITTED = 3,
  UNRESOLVED = 1
}

export default ResolutionStatus
