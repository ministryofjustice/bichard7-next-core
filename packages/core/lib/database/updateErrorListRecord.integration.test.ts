import createDbConfig from "@moj-bichard7/common/db/createDbConfig"
import ExceptionCode from "bichard7-next-data-latest/dist/types/ExceptionCode"
import postgres from "postgres"
import generateMockPhase1Result from "../../phase1/tests/helpers/generateMockPhase1Result"
import generateMockPhase2Result from "../../phase2/tests/helpers/generateMockPhase2Result"
import type ErrorListRecord from "../../types/ErrorListRecord"
import ResolutionStatus from "../../types/ResolutionStatus"
import errorPaths from "../exceptions/errorPaths"
import insertErrorListRecord from "./insertErrorListRecord"
import updateErrorListRecord from "./updateErrorListRecord"

const dbConfig = createDbConfig()
const db = postgres({
  ...dbConfig,
  types: {
    date: {
      to: 25,
      from: [1082],
      serialize: (x: string): string => x,
      parse: (x: string): Date => {
        return new Date(x)
      }
    }
  }
})

const snapshotExclusions = {
  error_id: expect.any(Number),
  trigger_insert_ts: expect.any(Date),
  msg_received_ts: expect.any(Date),
  error_insert_ts: expect.any(Date),
  create_ts: expect.any(Date),
  annotated_msg: expect.any(String),
  updated_msg: expect.any(String),
  message_id: expect.any(String)
}

describe("updateErrorListRecord", () => {
  beforeEach(async () => {
    await db`DELETE FROM br7own.error_list`
  })

  const testInputs = [
    { beforeStatus: null, exceptionsGenerated: false, expectedStatus: null },
    { beforeStatus: null, exceptionsGenerated: true, expectedStatus: ResolutionStatus.UNRESOLVED },
    {
      beforeStatus: ResolutionStatus.UNRESOLVED,
      exceptionsGenerated: true,
      expectedStatus: ResolutionStatus.UNRESOLVED
    },
    {
      beforeStatus: ResolutionStatus.UNRESOLVED,
      exceptionsGenerated: false,
      expectedStatus: ResolutionStatus.RESOLVED
    },
    {
      beforeStatus: ResolutionStatus.SUBMITTED,
      exceptionsGenerated: false,
      expectedStatus: ResolutionStatus.RESOLVED
    },
    {
      beforeStatus: ResolutionStatus.SUBMITTED,
      exceptionsGenerated: true,
      expectedStatus: ResolutionStatus.UNRESOLVED
    },
    { beforeStatus: ResolutionStatus.RESOLVED, exceptionsGenerated: true, expectedStatus: ResolutionStatus.UNRESOLVED },
    { beforeStatus: ResolutionStatus.RESOLVED, exceptionsGenerated: false, expectedStatus: ResolutionStatus.RESOLVED }
  ]

  testInputs.forEach(({ beforeStatus, exceptionsGenerated, expectedStatus }) => {
    const beforeStatusText = beforeStatus ? ResolutionStatus[beforeStatus] : "NULL"
    const expectedStatusText = expectedStatus ? ResolutionStatus[expectedStatus] : "NULL"
    const exceptionsText = exceptionsGenerated ? "generated" : "not generated"

    it(`should update the record and update error_status from ${beforeStatusText} to ${expectedStatusText} when message type is Phase 2 and exceptions ${exceptionsText}`, async () => {
      const result = generateMockPhase2Result()
      const recordId = await insertErrorListRecord(db, result)
      await db`UPDATE br7own.error_list SET error_status = ${beforeStatus}::integer WHERE error_id = ${recordId}`

      result.outputMessage.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant.ArrestSummonsNumber = "new_asn"
      result.outputMessage.AnnotatedHearingOutcome.HearingOutcome.Case.PTIURN = "new_ptiurn"
      result.outputMessage.AnnotatedHearingOutcome.HearingOutcome.Case.ForceOwner!.SecondLevelCode = "99"
      if (exceptionsGenerated) {
        result.outputMessage.Exceptions.push({ code: ExceptionCode.HO200100, path: errorPaths.case.asn })
      }

      await updateErrorListRecord(db, recordId, result)

      const updatedRecord = (
        await db<ErrorListRecord[]>`
        SELECT * FROM br7own.error_list WHERE error_id = ${recordId}`
      )[0]

      expect(updatedRecord.error_status).toBe(expectedStatus)
      expect(updatedRecord).toMatchSnapshot(snapshotExclusions)
    })

    it(`should update the record and update error_status from ${beforeStatusText} to ${expectedStatusText} when message type is Phase 1 and exceptions ${exceptionsText}`, async () => {
      const result = generateMockPhase1Result()
      const recordId = await insertErrorListRecord(db, result)
      await db`UPDATE br7own.error_list SET error_status = ${beforeStatus}::integer WHERE error_id = ${recordId}`

      result.hearingOutcome.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant.ArrestSummonsNumber = "new_asn"
      result.hearingOutcome.AnnotatedHearingOutcome.HearingOutcome.Case.PTIURN = "new_ptiurn"
      result.hearingOutcome.AnnotatedHearingOutcome.HearingOutcome.Case.ForceOwner!.SecondLevelCode = "99"
      if (exceptionsGenerated) {
        result.hearingOutcome.Exceptions.push({ code: ExceptionCode.HO100100, path: errorPaths.case.asn })
      }

      await updateErrorListRecord(db, recordId, result)

      const updatedRecord = (
        await db<ErrorListRecord[]>`
        SELECT * FROM br7own.error_list WHERE error_id = ${recordId}`
      )[0]

      expect(updatedRecord.error_status).toBe(expectedStatus)
      expect(updatedRecord).toMatchSnapshot(snapshotExclusions)
    })
  })
})
