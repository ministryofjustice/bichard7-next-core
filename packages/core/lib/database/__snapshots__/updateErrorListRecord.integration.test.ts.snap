// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateErrorListRecord should update the record and update error_status from NULL to NULL when message type is Phase 1 and exceptions not generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 0,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": null,
  "error_reason": null,
  "error_report": "",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": null,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 1,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from NULL to NULL when message type is Phase 2 and exceptions not generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 0,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": null,
  "error_reason": null,
  "error_report": "",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": null,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 2,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from NULL to UNRESOLVED when message type is Phase 1 and exceptions generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 1,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": 1,
  "error_reason": "HO100100",
  "error_report": "HO100100||br7:ArrestSummonsNumber",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 1,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 1,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from NULL to UNRESOLVED when message type is Phase 2 and exceptions generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 1,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": 1,
  "error_reason": "HO200100",
  "error_report": "HO200100||br7:ArrestSummonsNumber",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 1,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 2,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from RESOLVED to RESOLVED when message type is Phase 1 and exceptions not generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 0,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": null,
  "error_reason": null,
  "error_report": "",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 2,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 1,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from RESOLVED to RESOLVED when message type is Phase 2 and exceptions not generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 0,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": null,
  "error_reason": null,
  "error_report": "",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 2,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 2,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from RESOLVED to UNRESOLVED when message type is Phase 1 and exceptions generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 1,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": 1,
  "error_reason": "HO100100",
  "error_report": "HO100100||br7:ArrestSummonsNumber",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 1,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 1,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from RESOLVED to UNRESOLVED when message type is Phase 2 and exceptions generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 1,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": 1,
  "error_reason": "HO200100",
  "error_report": "HO200100||br7:ArrestSummonsNumber",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 1,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 2,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from SUBMITTED to RESOLVED when message type is Phase 1 and exceptions not generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 0,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": null,
  "error_reason": null,
  "error_report": "",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 2,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 1,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from SUBMITTED to RESOLVED when message type is Phase 2 and exceptions not generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 0,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": null,
  "error_reason": null,
  "error_report": "",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 2,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 2,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from SUBMITTED to UNRESOLVED when message type is Phase 1 and exceptions generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 1,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": 1,
  "error_reason": "HO100100",
  "error_report": "HO100100||br7:ArrestSummonsNumber",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 1,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 1,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from SUBMITTED to UNRESOLVED when message type is Phase 2 and exceptions generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 1,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": 1,
  "error_reason": "HO200100",
  "error_report": "HO200100||br7:ArrestSummonsNumber",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 1,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 2,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from UNRESOLVED to RESOLVED when message type is Phase 1 and exceptions not generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 0,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": null,
  "error_reason": null,
  "error_report": "",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 2,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 1,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from UNRESOLVED to RESOLVED when message type is Phase 2 and exceptions not generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 0,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": null,
  "error_reason": null,
  "error_report": "",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 2,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 2,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from UNRESOLVED to UNRESOLVED when message type is Phase 1 and exceptions generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 1,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": 1,
  "error_reason": "HO100100",
  "error_report": "HO100100||br7:ArrestSummonsNumber",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 1,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 1,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`updateErrorListRecord should update the record and update error_status from UNRESOLVED to UNRESOLVED when message type is Phase 2 and exceptions generated 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "new_asn",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_name_upper": "MAGISTRATES' COURTS LONDON CROYDON",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "defendant_name_upper": "TRTHREE TRPSTWO",
  "error_count": 1,
  "error_id": Any<Number>,
  "error_insert_ts": Any<Date>,
  "error_locked_by_id": null,
  "error_quality_checked": 1,
  "error_reason": "HO200100",
  "error_report": "HO200100||br7:ArrestSummonsNumber",
  "error_resolved_by": null,
  "error_resolved_ts": null,
  "error_status": 1,
  "is_urgent": 0,
  "last_pnc_failure_resubmission_ts": null,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "99ZD  ",
  "phase": 2,
  "pnc_update_enabled": "Y",
  "ptiurn": "new_ptiurn",
  "resolution_ts": null,
  "total_pnc_failure_resubmissions": 0,
  "trigger_count": 0,
  "trigger_insert_ts": Any<Date>,
  "trigger_locked_by_id": null,
  "trigger_quality_checked": null,
  "trigger_reason": null,
  "trigger_resolved_by": null,
  "trigger_resolved_ts": null,
  "trigger_status": null,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;
